lxc-to-lxd/main_migrate.go

```diff
@@ -0,0 +1,602 @@
+
+	return nil
+}
+
+func convertNetworkConfig(container *lxc.Container, devices types.Devices) error {
+	networkDevice := func(network map[string]string) map[string]string {
+		if network == nil {
+			return nil
+		}
+
+		device := make(map[string]string, 0)
+		device["type"] = "nic"
+
+		// Get the device type
+		device["nictype"] = network["type"]
+
+		// Convert the configuration
+		for k, v := range network {
+			switch k {
+			case "hwaddr", "mtu", "name":
+				device[k] = v
+			case "link":
+				device["parent"] = v
+			case "veth_pair":
+				device["host_name"] = v
+			case "":
+				// empty key
+				return nil
+			}
+		}
+
+		switch device["nictype"] {
+		case "veth":
+			_, ok := device["parent"]
+			if ok {
+				device["nictype"] = "bridged"
+			} else {
+				device["nictype"] = "p2p"
+			}
+		case "phys":
+			device["nictype"] = "physical"
+		case "empty":
+			return nil
+		}
+
+		return device
+	}
+
+	fmt.Println("Processing network configuration")
+
+	devices["eth0"] = make(map[string]string, 0)
+	devices["eth0"]["type"] = "none"
+
+	// New config key
+	for i := range container.ConfigItem("lxc.net") {
+		network := networkGet(container, i, "lxc.net")
+
+		dev := networkDevice(network)
+		if dev == nil {
+			continue
+		}
+
+		devices[fmt.Sprintf("convert_net%d", i)] = dev
+	}
+
+	// Old config key
+	for i := range container.ConfigItem("lxc.network") {
+		network := networkGet(container, i, "lxc.network")
+
+		dev := networkDevice(network)
+		if dev == nil {
+			continue
+		}
+
+		devices[fmt.Sprintf("convert_net%d", len(devices))] = dev
+	}
+
+	return nil
+}
+
+func convertStorageConfig(conf []string, devices types.Devices) error {
+	fmt.Println("Processing storage configuration")
+
+	i := 0
+	for _, mount := range getConfig(conf, "lxc.mount.entry") {
+		parts := strings.Split(mount, " ")
+		if len(parts) < 4 {
+			return fmt.Errorf("Invalid mount configuration: %s", mount)
+		}
+
+		// Ignore mounts that are present in LXD containers by default.
+		if shared.StringInSlice(parts[0], []string{"proc", "sysfs"}) {
+			continue
+		}
+
+		device := make(map[string]string, 0)
+		device["type"] = "disk"
+
+		// Deal with read-only mounts
+		if shared.StringInSlice("ro", strings.Split(parts[3], ",")) {
+			device["readonly"] = "true"
+		}
+
+		// Deal with optional mounts
+		if shared.StringInSlice("optional", strings.Split(parts[3], ",")) {
+			device["optional"] = "true"
+		} else {
+			if strings.HasPrefix(parts[0], "/") {
+				if !shared.PathExists(parts[0]) {
+					return fmt.Errorf("Invalid path: %s", parts[0])
+				}
+			} else {
+				continue
+			}
+		}
+
+		// Set the source
+		device["source"] = parts[0]
+
+		// Figure out the target
+		if !strings.HasPrefix(parts[1], "/") {
+			device["path"] = fmt.Sprintf("/%s", parts[1])
+		} else {
+			rootfs, err := getRootfs(conf)
+			if err != nil {
+				return err
+			}
+			device["path"] = strings.TrimPrefix(parts[1], rootfs)
+		}
+
+		devices[fmt.Sprintf("convert_mount%d", i)] = device
```

Comment:
Again with the `convert_*` prefix? 😁

Reply:
This was taken directly from the `lxc-to-lxd` script.

Reply:
Same, please change to `disk%d` or `mount%d`. Just looks cleaner to me.